src/core/mod.rs

```diff
@@ -21,34 +21,156 @@ use parsec_interface::operations::psa_key_attributes::KeyAttributes;
+///# use parsec_client::CoreClient;
+///# let client: CoreClient = CoreClient::new(AuthenticationData::AppIdentity(String::from("app-name")));
+///use parsec_client::core::ProviderID;
+///
+///let desired_provider = ProviderID::Tpm;
+///let available_providers = client.list_providers().expect("Failed to list providers");
+///if available_providers
+///    .iter()
+///    .filter(|provider| provider.id == desired_provider)
+///    .count()
+///    == 0
+///{
+///    panic!("Did not find desired provider!");
+///}
+///```
+///
+/// Checking operations supported by the provider we're interested in:
+///```no_run
+///# use parsec_client::auth::AuthenticationData;
+///# use parsec_client::CoreClient;
+///# let client: CoreClient = CoreClient::new(AuthenticationData::AppIdentity(String::from("app-name")));
+///# use parsec_client::core::ProviderID;
+///# let desired_provider = ProviderID::Tpm;
+///use parsec_client::core::Opcode;
+///let provider_opcodes = client
+///    .list_provider_operations(desired_provider)
+///    .expect("Failed to list opcodes");
+///// Each operation is identified by a specific `Opcode`
+///assert!(provider_opcodes.contains(&Opcode::PsaGenerateKey));
+///assert!(provider_opcodes.contains(&Opcode::PsaSignHash));
+///assert!(provider_opcodes.contains(&Opcode::PsaDestroyKey));
+///```
+///
+/// Creating a key-pair for signing SHA256 digests with RSA PKCS#1 v1.5:
+///```no_run
+///# use parsec_client::auth::AuthenticationData;
+///# use parsec_client::CoreClient;
+///# let client: CoreClient = CoreClient::new(AuthenticationData::AppIdentity(String::from("app-name")));
+///# use parsec_client::core::ProviderID;
+///# let desired_provider = ProviderID::Tpm;
+///use parsec_client::core::psa_algorithm::{Algorithm, AsymmetricSignature, Hash};
+///use parsec_client::core::psa_key_attributes::{KeyAttributes, KeyPolicy, KeyType, UsageFlags};
+///
+///let key_name = String::from("rusty key 🔑");
+///// This algorithm identifier will be used within the key policy (i.e. what
+///// algorithms are usable with the key) and for indicating the desired
+///// algorithm for each operation involving the key.
+///let asym_sign_algo = AsymmetricSignature::RsaPkcs1v15Sign {
+///    hash_alg: Hash::Sha256,
+///};
+///
+///// The key attributes define and limit the usage of the key material stored
+///// by the underlying cryptographic provider.
+///let key_attrs = KeyAttributes {
+///    key_type: KeyType::RsaKeyPair,
+///    key_bits: 2048,
+///    key_policy: KeyPolicy {
+///        key_usage_flags: UsageFlags {
+///            export: true,
+///            copy: true,
+///            cache: true,
+///            encrypt: false,
+///            decrypt: false,
+///            sign_message: true,
+///            verify_message: false,
+///            sign_hash: true,
+///            verify_hash: false,
+///            derive: false,
+///        },
+///        key_algorithm: Algorithm::AsymmetricSignature(asym_sign_algo),
```

Comment:
Just FYI: you can do `asym_sign_algo.into()` here 🤓

Reply:
Nice! Didn't know

Reply:
I found it a very nice pattern to do (implementing the `From`) for all enums that are wrappers around specific types!